packages/trader/src/AppV2/Hooks/useContractsForCompany.ts

```diff
@@ -28,9 +28,10 @@ const useContractsForCompany = () => {
     const [contract_types_list, setContractTypesList] = React.useState<TContractTypesList | []>([]);
 
     const [trade_types, setTradeTypes] = React.useState<TContractType[]>([]);
-    const { contract_type, onChange, setContractTypesListV2 } = useTraderStore();
+    const { contract_type, onChange, setContractTypesListV2, processContractsForV2, symbol } = useTraderStore();
     const { client } = useStore();
     const { loginid, is_switching, landing_company_shortcode } = client;
+    const prev_landing_company_shortcode_ref = React.useRef(landing_company_shortcode);
```

Comment:
You can also use usePrevious hook, it incapsulated the same logic, based on useRef.
Absolutely optional suggestion

Reply:
yeah ive tried to use before. doesnt work properly